### PR TITLE
docs2.0: Devtron2.0 doc changes in setup/install

### PR DESCRIPTION
**Pull request by divyabhushan**
# Description

Devtron2.0 doc changes in setup/install

## Type of change

- [x] This change requires a documentation update

# Checklist:

* [x] I've updated documentation as required by this PR.

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -6,7 +6,9 @@
 
 * [Setup](setup/install/setup.md)
   * [Installing Devtron](setup/install/README.md)
-    * [Install using Helm3](setup/install/install-devtron-helm-3.md)
+    * [Install Devtron](setup/install/install-devtron.md)
+    * [Install Devtron with CI/CD module](setup/install/install-devtron-with-cicd.md)
+    * [Install Devtron with Helm3](setup/install/install-devtron-helm-3.md)
```
Comment (prakarsh-dt) [inline docs/SUMMARY.md:11]: This was to be removed I guess as per our discussion?

File: docs/setup/install/README.md [hunk 2]
```diff
@@ -1,169 +1,48 @@
...
-| **DEX\_CONFIG** | dex config if you want to integrate login with SSO \(optional\) for more information check [Argocd documentation](https://argoproj.github.io/argo-cd/operator-manual/user-management/) |  |
-| **EXTERNAL\_SECRET\_AMAZON\_REGION** | AWS region for secret manager to pick \(required\) |  |
-| **PROMETHEUS\_URL** | url of prometheus where all cluster data is stored, if this is wrong, you will not be able to see application metrics like cpu, ram, http status code, latency and throughput \(required\) |  |
+* [Devtron](install-devtron.md): The base installation of Devtron includes basic Helm charts. Devtron base install helps to deploy, observe, manage, and debug existing Helm applications in all the clusters.
+* [Devtron with CI/CD](install-devtron-with-cicd.md): This mode includes the base install and the CI/CD module. The CI/CD module installation helps to perform CI/CD, security scanning, GitOps, Access control, debugging, and observability.
```
Comment (prakarsh-dt) [inline docs/setup/install/README.md:8]: We should rephrase and avoid using modes. We can write something like:

`The base installation of Devtron along with CI/CD module installed.`

File: docs/setup/install/README.md [hunk 3]
```diff
@@ -1,169 +1,48 @@
...
 
-#### Storage for Logs and Cache
+## Recommended resources
 
-AWS SPECIFIC
+The minimum requirements for Devtron basic and Devtron with CI/CD installation in production and non-production environments include:
```
Comment (prakarsh-dt) [inline docs/setup/install/README.md:12]: Remove `basic`, let's limit the phrases `base` , `basic` for discussions per se. In the documents, let's stick to Devtron and Devtron with CICD module.

File: docs/setup/install/README.md [hunk 4]
```diff
@@ -1,169 +1,48 @@
...
+ 
+Refer to the [Creating a Production grade EKS cluster using EKSCTL](https://devtron.ai/blog/creating-production-grade-kubernetes-eks-cluster-eksctl/) article to set up a cluster in the production environment.
 
+## Installing Devtron
+ 
+* [Install Devtron](install-devtron.md)
+* [Install Devtron with CI/CD module](install-devtron-with-cicd.md)
+* [Upgrade Devtron to Latest Version](#upgrade-devtron)
+ 
 ## Upgrade Devtron
```
Comment (prakarsh-dt) [inline docs/setup/install/README.md:42]: Remove this section from here, the commands specified here are no longer valid.

File: docs/setup/install/install-devtron-helm-3.md [hunk 5]
```diff
@@ -1,6 +1,15 @@
-# Install Devtron using Helm3
+# Install Devtron using Helm3 (Deprecated)
 
-To install Helm3, please check [Installing Helm3](https://helm.sh/docs/intro/install/)
+> Info: From Devtron v0.3.27 onwards, using Helm3 to install Devtron will be deprecated.
```
Comment (prakarsh-dt) [inline docs/setup/install/install-devtron-helm-3.md:3]: Installation using Helm3 is not deprecated. Using Helm2 and kubectl are deprecated. Helm3 is the only way to install devtron now.

We are not removing this page because we don't want users to land on error 404

File: docs/setup/install/install-devtron-with-cicd.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
+# Install Devtron with CI/CD module
+
+The Devtron installation includes the base installation and CI/CD modules.
```
Comment (prakarsh-dt) [inline docs/setup/install/install-devtron-with-cicd.md:3]: module as it is a single module

File: docs/setup/install/install-devtron-with-cicd.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+Please make sure that you do not have anything inside namespaces devtroncd, devtron-cd, devtron-ci, and devtron-demo as the below steps will clean everything inside these namespaces:
+
+```bash
+helm uninstall devtron --namespace devtroncd
+kubectl delete -n devtroncd -f https://raw.githubusercontent.com/devtron-labs/charts/main/charts/devtron/crds/crd-devtron.yaml
+kubectl delete -n argo -f https://raw.githubusercontent.com/devtron-labs/devtron/main/manifests/yamls/workflow.yaml
+kubectl delete ns devtroncd devtron-cd devtron-ci devtron-demo
+```
+
+### Cleanup
```
Comment (prakarsh-dt) [inline docs/setup/install/install-devtron-with-cicd.md:143]: This section is not required, the steps mentioned above shall do the cleanup.

File: docs/setup/install/install-devtron.md [hunk 8]
```diff
@@ -1,24 +1,20 @@
-# Install Hyperion using Helm3
```
Comment (prakarsh-dt) [inline docs/setup/install/install-devtron.md:1]: Why these changes? We can just mention `Deprecated` and tell the users that Hyperion is now Devtron.

File: docs/setup/install/installation-configuration.md [hunk 9]
```diff
@@ -1,29 +1,94 @@
...
+```
+
+> **Note**:
+> 1. Ensure that the **cluster has read and write access** to the S3 buckets/Azure Blob storage container mentioned in DEFAULT\_CACHE\_BUCKET, DEFAULT\_BUILD\_LOGS\_BUCKET or AZURE\_BLOB\_CONTAINER\_CI\_LOG, or AZURE\_BLOB\_CONTAINER\_CI\_CACHE.
+> 2. Ensure that the cluster has **read access** to AWS secrets backends \(SSM & secrets manager\).
+
+---
+
+The following tables contain parameters and their details for Secrets and ConfigMaps that are configured during the installation of Devtron. 
+While installing Devtron using `kubectl` the following parameters can be tweaked in [devtron-operator-configs.yaml](https://github.com/devtron-labs/devtron/blob/main/manifests/install/devtron-operator-configs.yaml) file. If the installation is proceeded using `helm3`, the values can be tweaked in [values.yaml](https://github.com/devtron-labs/charts/blob/main/charts/devtron/values.yaml) file.
```
Comment (prakarsh-dt) [inline docs/setup/install/installation-configuration.md:68]: We don't have an option to install using kubectl so we can remove that info from here

File: docs/setup/install/installation-configuration.md [hunk 10]
```diff
@@ -62,9 +126,6 @@ HIDE_DISCORD=false
...
 |-|-|
-|RECOMMEND_SECURITY_SCANNING | If True `security scanning` is `enabled` by default for a new build pipeline. Users can however turn it off in the new or existing pipelines.|
-|FORCE_SECURITY_SCANNING | If set True, `security scanning` is forcefully `enabled` by default for a new build pipeline. User can not turn it off for new as well as for existing build pipelines. Old pipelines that have security scanning disabled will remain unchanged and image scanning should be enabled manually for them.|
-|HIDE_DISCORD | Hides discord chat bot from dashboard.|
-
-
-
+|RECOMMEND_SECURITY_SCANNING | If True, `security scanning` is `enabled` by default for a new build pipeline. Users can however turn it off in the new or existing pipelines.|
+|FORCE_SECURITY_SCANNING | If set to True, `security scanning` is forcefully `enabled` by default for a new build pipeline. Users can not turn it off for new as well as for existing build pipelines. Old pipelines that have security scanning disabled will remain unchanged and image scanning should be enabled manually for them.|
+|HIDE_DISCORD | Hides discord chatbot from the dashboard.|
```
Comment (prakarsh-dt) [inline docs/setup/install/installation-configuration.md:131]: If set to true, hides discord chatbot from the dashboard, default is false.

**Comment by divyabhushan** [comment]
Updating the changes
